src/components/MovieDetails.tsx

```diff
@@ -1,9 +1,108 @@
-const MovieDetails = () => {
-  return (
-    <div>
-      <p>Movie details</p>
+import { useState, useEffect } from 'react';
+
+import api from 'src/service/api';
+
+import ReadMore from './ReadMore';
+
+import '../styles/MovieDetails.scss';
+
+type MovieDetailsProps = {
+  match: { params: { id: string } }
+}
+
+const MovieDetails = ({ match }: MovieDetailsProps) => {
+
+  const [movieId, setMovieId] = useState<string>(' ');
+  const [movieData, setMovieData] = useState({
+    poster_path: '',
+    original_title: '',
+    title: '',
+    tagline: '',
+    status: '',
+    release_date: '',
+    budget: '',
+    popularity: '',
+    vote_average: '',
+    runtime: '',
+    genres: [{ id: null, name: ' ' }],
+    production_countries: [{ name: '' }],
+    overview: '',
+  })
+
+  const {
+    poster_path,
+    original_title,
+    title,
+    tagline,
+    status,
+    release_date,
+    budget,
+    popularity,
+    vote_average,
+    runtime,
+    genres,
+    production_countries,
+    overview
+  } = movieData;
+
+  const poster = api.changeImgLinks(poster_path, 'w500');
+
+  const getMovieById = (id: number): void => {
+    api.getDataById(id).then((res: any) => {
+      setMovieData(res);
+    })
+  }
+
+  useEffect(() => {
+    setMovieId(match.params.id);
+    getMovieById(Number(movieId));
+  }, [movieId])
```

Comment:
hm, it looks like it might call getMovieById 2 times - first with ' space ', and second right away

Reply:
removed ' space ' in useState [movieId, setMovieId]

Reply:
Still cant get why not set setMovieId(match.params.id); on line 34? 
use effect is async operation - so it will call api with 0 (Number('')) returns zero first
here is how it looks in console: 
![Screenshot 2021-04-13 at 22 39 36](https://user-images.githubusercontent.com/16467053/114611022-3ce55280-9ca9-11eb-8181-82c50c1ffa4e.png)

the order of operations:
1 render component, set movieId to '', 
2 run useEffect, setNewMovieID
3 run getMovieById(Number(movieId)); with movieId that was set before on step 1
4 since we set movieId our use effect will run again
5  run useEffect, setNewMovieID (likely it is primitive, it would break for objects this way)
6 run getMovieById(Number(movieId)); with movieId that we need

Reply:
got it! maybe is it better to make an api request inside of useState with dependencies [] and not to store match.params.id at all?

Reply:
inside of useState with  >> inside of useEffect with 
yes, correct